Sources/LeafKit/LeafLexer.swift

```diff
@@ -5,12 +5,23 @@ extension Character {
     }
     
     var isValidInParameter: Bool {
-        return self.isLowercaseLetter
-            || self.isUppercaseLetter
+        return self.isValidInTagName
             || self.isValidOperator
-            || (.zero ... .nine) ~= self
-            || self == .period
+            || self.isValidInNumeric
+    }
+       
+    var canStartNumeric: Bool {
+        return (.zero ... .nine) ~= self
+    }
+    
+    var isValidInNumeric: Bool {
+        return self.canStartNumeric
             || self == .underscore
+            || self == .binaryNotation
+            || self == .octalNotation
+            || self == .hexNotation
+            || self.isHexidecimal
+            || self == .period
```

Comment:
Granularity changes to Operator observers

Reply:
Super nit-picky (and likely _extremely minor_ optimization) but maybe move `.period` forward in the grouping, as it will short-circuit more quickly than its current predecessor (though also depends on probability distribution... thinking to position 2 or 3?).

Reply:
Possibly but the test is for both Int/Double lexing, so it will never be hit for Ints, and at most once for something that IS a Double, assuming that the input is correctly formed (and if it isn't, efficiency isn't going to change much).

This is by no means an ideal way of lexing stuff but I'm sticking within the existing behavior pattern for the moment :)